cmd/kubeadm/app/cmd/phases/preflight.go

```diff
@@ -56,6 +55,52 @@ var (
 	errorMissingConfigFlag = errors.New("the --config flag is mandatory")
 )
 
+// preflightMasterContext defines the behavior that a runtime context passed to the PreflightMaster master phase
+// should have. Please note that we are using an interface in order to make this phase reusable in different workflows
+// (and thus with different runtime context, all of them requested to be compliant to this interface)
+type preflightMasterContext interface {
+	Cfg() *kubeadmapi.InitConfiguration
+	DryRun() bool
+	IgnorePreflightErrors() sets.String
+}
+
+// PreflightMaster defines that kubeadm workflow phase that implements preflight checks for a new master node.
+var PreflightMaster = workflow.SimplePhase{
+	Name:    "preflight",
+	Short:   "Run master pre-flight checks",
+	Long:    "Run master pre-flight checks, functionally equivalent to what implemented by kubeadm init.",
+	Example: masterPreflightExample,
+	Run:     runPreflightMaster,
+}
+
+// runPreflightMaster executes preflight checks logic.
+func runPreflightMaster(c workflow.RunnerContext) error {
+	ctx, ok := c.(preflightMasterContext)
+
+	if !ok {
+		return fmt.Errorf("preflight phase invoked with an invalid context")
+	}
+
+	fmt.Println("[preflight] running pre-flight checks")
+	if err := preflight.RunInitMasterChecks(utilsexec.New(), ctx.Cfg(), ctx.IgnorePreflightErrors()); err != nil {
+		return nil
+	}
+
+	// TODO: consider if to split this into a nested phase preflight/pull
+	if !ctx.DryRun() {
+		fmt.Println("[preflight] Pulling images required for setting up a Kubernetes cluster")
+		fmt.Println("[preflight] This might take a minute or two, depending on the speed of your internet connection")
+		fmt.Println("[preflight] You can also perform this action in beforehand using 'kubeadm config images pull'")
+		if err := preflight.RunPullImagesCheck(utilsexec.New(), ctx.Cfg(), ctx.IgnorePreflightErrors()); err != nil {
+			return err
+		}
+	} else {
+		fmt.Println("[preflight] Would pull the required images (like 'kubeadm config images pull')")
```

Comment:
It would be useful to dump the images that were supposed to be pulled here.

Reply:
i think we already dump them with --v>=0
no?

Reply:
I don't think so, but I am not 100% certain. We have to check on that.

Once this gets merged, I'll file an issue.

Reply:
--v=1 seems to show them:
```
[preflight/images] Pulling images required for setting up a Kubernetes cluster
[preflight/images] This might take a minute or two, depending on the speed of your internet connection
[preflight/images] You can also perform this action in beforehand using 'kubeadm config images pull'
I1017 18:33:05.090825    8365 checks.go:840] pulling k8s.gcr.io/kube-apiserver:v1.12.1
I1017 18:33:25.091417    8365 checks.go:840] pulling k8s.gcr.io/kube-controller-manager:v1.12.1
...
```
probably fine to leave it like that, unless we want to show them with v=0 (aka fmt.Print).

Reply:
That seems to be the case where there is no `--dry-run`. My idea is to have this with `--dry-run`.